pkg/filterengine/filters/object_annotation_checker.go

```diff
@@ -82,6 +107,101 @@ func isObjectNotifDisabled(obj metaV1.ObjectMeta) bool {
 	return false
 }
 
+// isObjectNamespaceNotifEnabled checks annotation botkube.io/notification on the namespace
+// annotation botkube.io/notification enable/disable all object's event notification in the namespace
+func isObjectNamespaceNotifEnabled(obj metaV1.ObjectMeta) bool {
+
+	namespace := obj.Namespace
+	nsObjectNotifEnabled, found := NamespaceCache.Get(namespace)
+
+	if !found {
+		log.Warnf("Cache Miss!! Namespace '%s' not found in the cache!", namespace)
+
+		// get namespace object
+		nsObject, err := utils.DynamicKubeClient.Resource(schema.GroupVersionResource{Group: "", Version: "v1", Resource: "namespaces"}).Get(context.TODO(), namespace, v1.GetOptions{})
+
+		if err != nil {
+			log.Error(err)
+			nsObjectNotifEnabled = "false"
+		} else {
```

Comment:
Same here. return early and get rid of else block

Reply:
How to manage common code which has to be executed in either case. Should we wrap it in both if and else condition the same piece of code.